### PR TITLE
Handle byte array property editing like we do for string values

### PR DESCRIPTION
**Pull request by TheOneRing**
Task-Id: KDEND-192

### CODE REVIEW

File: ui/propertyeditor/propertyeditorfactory.cpp [hunk 1]
```diff
@@ -45,6 +45,7 @@ PropertyEditorFactory::PropertyEditorFactory()
     initBuiltInTypes();
 
     addEditor(QVariant::Color, new QStandardItemEditorCreator<PropertyColorEditor>());
+    addEditor(QVariant::ByteArray, new QStandardItemEditorCreator<PropertyTextEditor>());
```
Comment (pasnox) [inline ui/propertyeditor/propertyeditorfactory.cpp]: I did check nothing but is not the editor would change the QByteArray type to a QString here ?

Comment (TheOneRing) [inline ui/propertyeditor/propertyeditorfactory.cpp]: I'd expect the QVariant magic to get the correct type.
I tested it with fragmentShader and the changes where applied as expected.

Comment (pasnox) [inline ui/propertyeditor/propertyeditorfactory.cpp]: Hm what if the QString is not ascii encoded ? does it keep the correct chars like éàç ?

File: ui/propertyeditor/propertytexteditor.cpp [hunk 2]
```diff
@@ -52,39 +52,42 @@ QString PropertyTextEditorDialog::text() const
...
-    // The user already pressed Apply, don't force her/him to do again
-    QKeyEvent event(QEvent::KeyPress, Qt::Key_Enter, Qt::NoModifier);
-    QApplication::sendEvent(this, &event);
 }
 
-void PropertyTextEditor::edit()
+void PropertyByteArrayEditor::edit()
 {
-    PropertyTextEditorDialog dlg(text(), this);
+    PropertyTextEditorDialog dlg(value().toByteArray(), this);
```
Comment (pasnox) [inline ui/propertyeditor/propertytexteditor.cpp]: Let be clear we assume all utf8 by default and for Qt4 support, let use QString::fromUtf8()

**Pull request merged**